q2templates/_templates.py

```diff
@@ -5,3 +5,58 @@
 #
 # The full license is in the file COPYING.txt, distributed with this software.
 # ----------------------------------------------------------------------------
+
+import os
+import pkg_resources
+import shutil
+import tempfile
+
+from .util import copy_assets, get_iterable
+from jinja2 import Environment, FileSystemLoader
+
+
+def render(source_files, output_dir, styles=None, context=None):
+    """Render user provided source files into a QIIME 2 visualization template.
+
+
+    Parameters
+    ----------
+    source_files : str, or list of str
+        Files to be rendered and written to the output_dir.
+    output_dir : str
+        The output_dir provided to a visualiation function by the QIIME 2
+        framework.
+    styles : str or list of str, optional
+        The necessary QIIME 2 templates that are used to render the
+        source_files.
+    context : dict, optional
+        The context dictionary to be rendered into the source_files.
```

Comment:
Can you add a note here that the same context is passed to all templates to be rendered? Thanks for adding the docstring, super awesome!